streaming/src/main/scala/org/apache/spark/streaming/util/RecurringTimer.scala

```diff
@@ -87,22 +87,32 @@ class RecurringTimer(clock: Clock, period: Long, callback: (Long) => Unit, name:
     prevTime
   }
 
+  private def triggerActionForNextInterval(): Unit = {
```

Comment:
Could you update the docs on stop() to capture the information that it is guaranteed that there will be at least one callback after the stop has been called.